react-components/src/data-providers/core-dm-provider/filterNodesByMappedTo3d.ts

```diff
@@ -0,0 +1,251 @@
+  {
+    source: COGNITE_CAD_NODE_SOURCE,
+    properties: ['object3D']
+  }
+] as const satisfies SourceSelectorV3;
+
+const checkEquipmentFilter = {
+  with: {
+    initial_nodes: {
+      nodes: {
+        filter: {
+          and: [
+            {
+              in: {
+                property: ['node', 'externalId'],
+                values: { parameter: 'initialExternalIds' }
+              }
+            },
+            {
+              hasData: [COGNITE_VISUALIZABLE_SOURCE]
+            }
+          ]
+        }
+      }
+    },
+    directly_referenced_nodes: {
+      nodes: {
+        filter: {
+          and: [
+            {
+              in: {
+                property: ['node', 'externalId'],
+                values: { parameter: 'directlyMappedIds' }
+              }
+            },
+            {
+              hasData: [COGNITE_VISUALIZABLE_SOURCE]
+            }
+          ]
+        }
+      }
+    },
+    indirectly_referenced_edges: {
+      edges: {
+        from: 'initial_nodes',
+        direction: 'outwards',
+        nodeFilter: {
+          hasData: [COGNITE_VISUALIZABLE_SOURCE]
+        }
+      }
+    },
+    indirectly_referenced_nodes: {
+      nodes: {
+        from: 'indirectly_referenced_edges'
+      }
+    },
+    initial_nodes_object_3ds: getObject3dRelation('initial_nodes'),
+    initial_nodes_cad_nodes: getRevisionsCadNodeFromObject3D('initial_nodes_object_3ds'),
+    initial_nodes_point_cloud_volumes: getRevisionsPointCloudVolumes('initial_nodes_object_3ds'),
+    direct_nodes_object_3ds: getObject3dRelation('directly_referenced_nodes'),
+    direct_nodes_cad_nodes: getRevisionsCadNodeFromObject3D('direct_nodes_object_3ds'),
+    direct_nodes_point_cloud_volumes: getRevisionsPointCloudVolumes('direct_nodes_object_3ds'),
+    indirect_nodes_object_3ds: getObject3dRelation('indirectly_referenced_nodes'),
+    indirect_nodes_cad_nodes: getRevisionsCadNodeFromObject3D('indirect_nodes_object_3ds'),
+    indirect_nodes_point_cloud_volumes: getRevisionsPointCloudVolumes('indirect_nodes_object_3ds')
+  },
+  select: {
+    indirectly_referenced_edges: {},
+    initial_nodes_object_3ds: { sources: cogniteObject3dSourceWithProperties },
+    initial_nodes_cad_nodes: {
+      sources: cadNodeSourceWithProperties
+    },
+    initial_nodes_point_cloud_volumes: { sources: pointCloudVolumeSourceWithProperties },
+    direct_nodes_object_3ds: { sources: cogniteObject3dSourceWithProperties },
+    direct_nodes_cad_nodes: {
+      sources: cadNodeSourceWithProperties
+    },
+    direct_nodes_point_cloud_volumes: { sources: pointCloudVolumeSourceWithProperties },
+    indirect_nodes_object_3ds: { sources: cogniteObject3dSourceWithProperties },
+    indirect_nodes_cad_nodes: {
+      sources: cadNodeSourceWithProperties
+    },
+    indirect_nodes_point_cloud_volumes: {
+      sources: pointCloudVolumeSourceWithProperties
+    }
+  }
+} as const satisfies Omit<QueryRequest, 'parameters' | 'cursor'>;
```

Comment:
Is it possible (and if does make sense) to move out the consts and the type definitions? Or at least organize into a separated place inside the file. I think moving out we can reduce the file size.

Reply:
I agree, the file was becoming a bit too big